DNN Platform/Library/UI/Skins/SkinPackageInfo.cs

```diff
@@ -27,6 +28,7 @@ public class SkinPackageInfo : BaseEntityInfo, IHydratable
         private string skinType;
         private Dictionary<int, string> skins = new Dictionary<int, string>();
 
+        /// <inheritdoc cref="ISkinPackageInfo.PackageId"/>
         public int PackageID
```

Comment:
Do we want to mark these properties as deprecated?

Reply:
I'm against this if `int PackageId` is not explicit.

This happend to PortalId as well:

https://github.com/dnnsoftware/Dnn.Platform/blob/3af88c5078e102d2316aac90a72c0abb634882f3/DNN%20Platform/Library/Entities/Portals/PortalInfo.cs#L212-L221

`ID` was deprecated but this makes it really awkward for current code. If you try to get the portal ID, you get a deprecated warning:

```cs
portalInfo.PortalID; // Deprecate warning.
```

But if you try to get `PortalId`, you'll get an error:
```cs
portalInfo.PortalId; // Error: PortalId does not exists.
```

This is because `PortalId` is an implicit implementation, so you have to cast the instance to an interface before you can get it (because `ThisAsInterface` is private):
```cs
((IPortalInfo)portalInfo).PortalID;
```

---

I suggest the following:

Add `[Obsolete]` and `[EditorBrowsable]` to the property, and make `PackageId` public:
```cs
public class SkinPackageInfo
{
    [Obsolete("Deprecated in DotNetNuke 10.0.0. Use DotNetNuke.Abstractions.Portals.IPortalInfo.PortalId instead. Scheduled removal in v12.0.0.")]
    [EditorBrowsable(EditorBrowsableState.Never)]
    public int PackageID
    {
        get => this.PackageId;
        set => this.PackageId = value;
    }
    
    public int PackageId
    {
        get => this.packageID;
        set => this.packageID = value;
    }
}
```

This way:
1. You can still access `skinPackageInfo.PackageID` with a deprecated warning.
2. You can change it to `skinPackageInfo.PackageId` without an error.
3. You won't see `PackageID` and `PackageId` in Visual Studio, since the property `PortalID` is hidden.
    _Note, in [Rider](https://www.jetbrains.com/rider/) the property is still visible [if the setting in "Settings → Editor → General → Code Completion → Filter members by [EditorBrowsable] attribute" isn't enabled](https://youtrack.jetbrains.com/issue/RIDER-17669/Respect-EditorBrowsableEditorBrowsableState.Never#focus=Comments-27-6455778.0-0)._

Reply:
The problem with this is that usage of the property becomes ambiguous in VB.NET, since it's case insensitive (this was what we originally did with `PortalInfo.PortalId`, until users reported errors https://github.com/dnnsoftware/Dnn.Platform/pull/4397).

I agree that it's awkward to deprecate the member without having the replacing member visible. I don't have strong feelings about deprecating the property or not. Any thoughts @dnnsoftware/approvers?

Reply:
I didn't know VB.NET was case insensitive...

So either VB.NET does a cast (`DirectCast(info, ISkinPackageInfo).PackageId`) when we make it public or C# does a cast (`((IPortalInfo)ISkinPackageInfo).PackageId`) when we make it an implicit implementation 😅.

Both are annoying, but VB.NET will not compile and is being used a lot in skins; which will break a lot of skins.
C# is just a warning, still annoying if you want to fix all the deprecations but it won't break anything.

In my opinion the best thing to do is to not deprecate the property.

Reply:
Unfortunately, we decided that our best option was to continue down the route we've already started, and live with pain of casting to the interface until we can remove the conflicting properties in DNN v11. We're not especially happy with it as a resolution either, but think it's the best long-term option for the platform. We really appreciate your input and contributions.